src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { useDeleteTodo } from '../../hooks/useDeleteTodo';
+import { useTodos } from '../../utils/TodoContext';
+import { ErrorType } from '../../types/ErrorType';
+
+type TodoItemProps = {
+  todo: Todo;
+  isTemp?: boolean;
+  onErrorChange: (error: ErrorType | null) => void; // Add onErrorChange prop
+};
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  todo,
+  isTemp,
+  onErrorChange,
+}) => {
+  const { deleteTodo, isDeleting, error } = useDeleteTodo(); // Get error from useDeleteTodo
+  const [showLoader, setShowLoader] = useState<boolean>(false);
```

Comment:
you don't need to specify type because TypeScript can determine it from initial value